core/platform/datastore/gae_datastore_services_test.py

```diff
@@ -144,3 +145,14 @@ def test_returns_value_of_nested_transaction(self):
             gae_datastore_services.transaction(
                 lambda: gae_datastore_services.transaction(lambda: 1)),
             1)
+
+
+class EnforcedPropertyTests(unittest.TestCase):
```

Comment:
Why not import from test_utils.GenericTestBase like the other classes (so that we have just one pattern throughout the codebase)?

Reply:
Because this specific pattern provides negative value, in my opinion. GenericTestBase is very heavy and performs many unnecessary operations (signing up a user, stubbing the entire testapp, etc.), but all that's needed by this test is the assertion methods. For this reason, I've left it out.

Reply:
So should we also update other test classes for which the above issues hold?

Reply:
Yes I think so, in general I think we should be more disciplined in what kinds of base classes tests inherit from- don't use what you don't need.

I don't think this is something that's particularly easy to retroactively fix, but at the very least we can incrementally catch things at review-time.

Reply:
I am still concerned about the introduction of a new convention here. If we have two ways of doing things in the codebase, then folks are going to copy one of them and get confused when it doesn't work. So I have a strong preference for there being just one way to do things, where possible.

Is there actually any developer-visible effect when the existing pattern is used? E.g. do the tests take longer to run? If not, I think we shouldn't do a one-off like this -- it would be better to just use the existing base class for now and file an issue for a later refactor across the general test suite.

Reply:
The tests do take longer to run, but it's more of a "death by a thousand cuts" sort of penalty. I really feel like we shouldn't be encouraging this only being it's easy to copy, we should be more careful about how the tests are structured (especially considering how long the tests take, 3+ hours).

Reply:
How much longer?

To be clear, though, I'm not objecting to the change. But such changes need to be managed properly. It doesn't just involve changing one individual instance in a new PR -- it entails documenting the policy in the "writing backend tests" wiki page, fixing other cases so that there is one convention in the codebase, doing something to prevent violations if possible (may not be possible in this case), and making sure that other folks know about it.

So, as mentioned above, I'm happy for this to be done in a separate refactor, especially if it doesn't make sense to bundle all of the above into this PR.

Reply:
4.6 seconds vs 3.54 seconds (5 runs each), so around 130% increase (again, death by a thousand cuts).

I'm not convinced this is something that can just be fixed by a refactor, it's something that needs to be enforced by reviewers as PRs come in ("Don't use what you don't need"). This is something that's too easy to forget about, and simple enough to explain to new contributors that it would not cause confusion. Not to mention, there isn't a lot of bandwidth to handle this, and I don't want to let it go in just for it to be forgotten...

Currently there are three usages of `unittest.TestCase` in the codebase (https://github.com/oppia/oppia/search?q=unittest.TestCase), so it's not the first PR to do it. I'd prefer we use this PR as set an example of "how to tell when a test actually needs GenericTestBase," and let reviewers refer to it when they want to flag improper inheritances. It serves as a good sample because the other test file, exp_jobs_one_off_test.py, _does_ require the GenericTestBase and all of the extra setup it performs.

WDYT?

Reply:
Yup, I buy it! Could you please update the ["writing backend tests" wiki page](https://github.com/oppia/oppia/wiki/Writing-backend-tests) with instructions for when to use which type of base class? You can point to this PR and some of the other instances you linked. 

Everything else sounds good, thanks for explaining. I wasn't aware there were other usages already.

Reply:
Awesome, done: https://github.com/oppia/oppia/wiki/Writing-backend-tests/_compare/b31176f4842918516ab028c57ffc8d1ac0f6a653...c10ac6bab86658f82d0ac97cc75578d99c26742b.